overlord/aspectstate/aspectstate.go

```diff
@@ -0,0 +1,122 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package aspectstate
+
+import (
+	"errors"
+
+	"github.com/snapcore/snapd/aspects"
+	"github.com/snapcore/snapd/overlord/aspectstate/aspecttest"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// Set finds the aspect identified by the account, bundleName and aspect and sets
+// the specified field to the supplied value.
+func Set(st *state.State, account, bundleName, aspect, field string, value interface{}) error {
```

Comment:
Super nitpick thought: 
If aspect-bundle param is `bundleName`, should aspect-field param not be `fieldName` if both are identified with `name: <...>` in the assertion?

Spec section `Aspects assertion header` shows example of unique "aspect view path" `<account-id>/<aspect-bundle>/<aspect>/<name>`, which perhaps could be `<account-id>/<aspect-bundle>/<aspect>/<field>`?

Reply:
That comes from this comment by Samuele https://github.com/snapcore/snapd/pull/12736#discussion_r1172510266. It only applied to  bundle, so I'm inclined to leave the others as they are as I think the short names are clear enough. About the error's field, I'll change it to be consistent. Thanks

About the "field" name, that's a good point, it isn't super consistent. As Samuele mentioned here https://github.com/snapcore/snapd/pull/12824#discussion_r1200606198 we might need to realign the naming once he comes back